### DISCUSSION TITLE
Problem with shelly RGBW2 external input

### DISCUSSION THREAD

**Discussion by BartoszGan** [RavenSystem/esp-homekit-devices#2189] [General]
Hi after updating to 12.10.0 the external switch on the shelly RGBW2 stopped working. Does anyone have an idea how to fix this? Below is the current configuration.

`{
    "c": {
        "io": [
            [
                [
                    12,
                    15
                ],
                7
            ],
            [
                [
                    14,
                    4
                ],
                7,
                0,
                0,
                2
            ],
            [
                [
                    2
                ],
                2
            ],
            [
                [
                    13
                ],
                6
            ],
            [
                [
                    5
                ],
                6,
                0,
                3
            ]
        ],
        "c": 1,
        "f": 20,
        "l": 2,
        "b": [
            {
                "g": 13,
                "t": 5
            }
        ]
    },
    "a": [
        {
            "t": 30,
            "g": [
                12,
                15,
                14,
                4
            ],
            "ld": [
                0,
                1,
                0,
                1
            ],
            "b": [
                {
                    "g": 5,
                    "p": 0,
                    "i": 1,
                    "t": 0,
                    "m": 1
                },
                {
                    "g":5,
                    "t":3
                }
            ],
            "es": [
                {
                    "i": 5,
                    "b": [
                        {
                            "g": 5,
                            "p": 0,
                            "i": 1,
                            "t": 3,
                            "v": 0
                        }
                    ]
                },
                {
                    "i": 5,
                    "b": [
                        {
                            "g": 5,
                            "p": 0,
                            "i": 1,
                            "t": 2,
                            "v": 0
                        }
                    ]
                }
            ]
        }
    ]
}`

**Comment by RavenSystem** [comment]
It is a bug. Stay tuned to this discussion.

**Reply by RavenSystem** [reply]
Update to new version.

**Comment by BartoszGan** [comment]
After uploading the newer version it is better, but sometimes the external button does not work. It is random. I suspect this is a configuration problem but I can't seem to get around it. Below is the current configuration:
`{ "c": { "io": [ [ [ 12, 15 ], 7 ], [ [ 14, 4 ], 7, 0, 0, 2 ], [ [ 2 ], 2 ], [ [ 13 ],6,1], [ [ 5 ], 6, 0, 1,1] ], "c": 1,  "f": 20, "l": 2, "b": [ { "g": 13, "t": 5 } ] }, "a": [ { "t": 30, "g": [ 12, 15, 14, 4 ], "ld": [ 0, 1, 0, 1 ], "b": [ { "g": 5, "p": 0, "i": 1, "t": 0, "m": 1 }, { "g":5, "t":3 } ], "es": [ { "i": 5, "b": [ { "g": 5, "p": 0, "i": 1, "t": 3, "v": 0 } ] }, { "i": 5, "b": [ { "g": 5, "p": 0, "i": 1, "t": 2, "v": 0 } ] } ] } ] }`

**Reply by RavenSystem** [reply]
Update to v12.12.0 and use your original script.

**Reply by BartoszGan** [reply]
After the upgrade, unfortunately no change:(
I would just like to add that I am using a bell switch. I don't know if this makes a difference.
